test/services/return-logs/fetch-download-return-log.service.test.js

```diff
@@ -0,0 +1,126 @@
+'use strict'
+
+// Test framework dependencies
+const Lab = require('@hapi/lab')
+const Code = require('@hapi/code')
+const Sinon = require('sinon')
+
+const { describe, it, beforeEach, afterEach } = (exports.lab = Lab.script())
+const { expect } = Code
+
+// Test helpers
+const ReturnLogHelper = require('../../support/helpers/return-log.helper.js')
+const ReturnSubmissionHelper = require('../../support/helpers/return-submission.helper.js')
+const ReturnSubmissionModel = require('../../../app/models/return-submission.model.js')
+const ReturnSubmissionLineHelper = require('../../support/helpers/return-submission-line.helper.js')
+
+// Thing under test
+const FetchDownloadReturnLogService = require('../../../app/services/return-logs/fetch-download-return-log.service.js')
+
+describe('Fetch Download Return Log service', () => {
+  let returnLog
+  let returnSubmissions = []
+
+  beforeEach(async () => {
+    returnLog = await ReturnLogHelper.add()
+
+    // We stub on the model prototype so that any created instances have $applyReadings stubbed. We don't set any return
+    // value as we don't need it to actually do anything; we just want to be able to assert that it was called.
+    Sinon.stub(ReturnSubmissionModel.prototype, '$applyReadings')
+  })
+
+  afterEach(() => {
+    Sinon.restore()
+  })
+
+  describe('when the return log with submissions exists', () => {
```

Comment:
Again, we should only be calling this service because we have displayed the download button, and we've only displayed that because the return log has submissions.

So, I think you can just focus the 'narrative' in these tests to be 'when a matching return log exists / when no matching return log exists'.